Wip/travis

madsflensted opened the pull request:
Ok - so I got Travis to [build and run the tests](https://travis-ci.org/madsflensted/xelm/builds/86248954). 

And then I realized that you had intended the users to run the tests [via the reactor](https://github.com/justalisteningman/xelm#running-tests), so my approach is maybe not what you were looking for.

And since this is a forked repository, I can't open and issue where we can discuss the setup, so I opened this PR instead.

I haven't done the Haskell exercises, but expect them to be all code and no GUI, so the same will probably be true for all the initial elm exercises. So using the elm-test module to compile and run tests seems like a simpler way for people to get started, without any complexities from installing jsdom, [apparently there can be problems on Windows](https://github.com/deadfoxygrandpa/Elm-Test#testing-from-the-command-line)

On the other hand the easy of feedback from the elm-reactor can also be a nice thing.

madscoaducom committed:
Test run locally

madscoaducom committed:
'config.json does not include [node_modules]' -> ignore node_modules

justalisteningman commented:
No, this is great!  The travis build is just for sanity for the contributors - it's intended to catch regressions.  The Reactor is intended for the end-users of exercism, as well as contributors as they make their own tests.

madsflensted commented:
Ok, du notice that I had to comment out the "main =..." from the three
tests, to avoid the jsdom dependency.

So something that needs fixing along the lines of including the example
file for testing.

I'll see if I can come up with something, but I guess that can be solved
down line.
On Oct 20, 2015 18:56, "Ben Watts" <notifications@github.com> wrote:

> No, this is great! The travis build is just for sanity for the
> contributors - it's intended to catch regressions. The Reactor is intended
> for the end-users of exercism, as well as contributors as they make their
> own tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/justalisteningman/xelm/pull/1#issuecomment-149631920>.
>

justalisteningman commented:
So, off the top of my head, there are a few ways to proceed: 
1. Abandon using the reactor for end-users, switch to a CLI-based approach.
2. Make a script that runs on travis that removes the main lines from the source files before they run on Travis.
3. Something else reasonable?

@madsflensted - can you speak to the UX (or even better, give us a screenshot) of option 1?  I've mainly used the reactor for my personal testing.  I'd like to not add any other dependencies (other than elm) for the end-user.  The simplicity of the browser-based approach appeals to me, given that this is a tool for new language users.

madsflensted commented:
I agree that the primary path should be no dependencies beyond elm itself, and option 2 should be do-able. And once the promised update of the reactor is released, that experience will be a super experience, and would allow us to add tests with graphical output.

When I get some time I will be happy to look into ways to do option 2

As Exercism it self uses the command line, I would probably suggest to have section in the readme, that shows users how to setup the node base elm test package, for those who want to go that way. Again will be happy to contribute that section.

madscoaducom committed:
Support CI and using elm-reactor

- Tag lines that needs enabling/disabling for CI to work
- Add CI test runner script
- Update all *Test.elm to use this format
- Add empty files for actual user to start with
- Add all existing examples to TestRunner

madscoaducom committed:
Travis: try to fix compile problem by pinning npm elm

madscoaducom committed:
Pin elm-test as well to fix problems with unset 'responses'

madsflensted commented:
Ok - I have made a version that supports both Travis and using the reactor for working with the exercises.

The ci-run-tests.sh script now enables/disables the necessary lines inside the *Test.elm files to be able to run the tests.  It basically looks for inline comments of `-- CI_ENABLE` or `-- CI_DISABLE`.

I  also added the empty exercise files, so the current exercises can be worked on.
